Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalMaterialShaderRef.cpp

```diff
@@ -198,6 +206,13 @@ void plMetalMaterialShaderRef::EncodeArguments(MTL::RenderCommandEncoder* encode
             return postEncodeTransform(layer, index);
         }
     );
+    
+    if (fBumps[pass].has_value())
+    {
+        plLayerInterface *bumpLayer = fMaterial->GetLayer(fMaterial->GetNumLayers()-1);
+        auto texture = (plMetalTextureRef*)bumpLayer->GetTexture()->GetDeviceRef();
+        encoder->setFragmentTexture(texture->fTexture, 7);
```

Comment:
I might need to look into this. Metal supports 8 texture passes - so I'm shoving the normal map into the last slot for convenience. But there are comments that imply multiple texture maps are possible - so I might need something more nuanced. Also hardcoded number bad.

Reply:
Yeah, I was thinking about that. If bump maps could be unlimited I would have to break each out into its own pass.

There is a comment somewhere else that hints multiple bump maps could be used for water. I haven't seen that used anywhere, but I guess I should could against that.

I'd probably keep support for single pass normal maps in the shader in since we've discussed having that available - but I'd have to separate out the glow maps a bit more to get them into proper additional stages.